libdnf/base/goal.cpp

```diff
@@ -1030,6 +1061,74 @@ void Goal::Impl::add_up_down_distrosync_to_goal(
     }
 }
 
+GoalProblem Goal::Impl::add_group_install_to_goal(
+    base::Transaction & transaction,
+    const std::string & spec,
+    const transaction::TransactionItemReason reason,
+    GoalJobSettings & settings) {
+    auto & cfg_main = base->get_config();
+    bool strict = settings.resolve_strict(cfg_main);
+    sack::QueryCmp cmp = settings.ignore_case ? sack::QueryCmp::IGLOB : sack::QueryCmp::GLOB;
+    comps::GroupQuery group_query(base, true);
+    if (settings.group_with_id) {
+        comps::GroupQuery group_query_id(base);
+        group_query_id.filter_groupid(spec, cmp);
+        group_query |= group_query_id;
+    }
+    if (settings.group_with_name) {
+        comps::GroupQuery group_query_name(base);
+        group_query_name.filter_name(spec, cmp);
+        group_query |= group_query_name;
+    }
+    if (group_query.empty()) {
+        auto problem = transaction.p_impl->report_not_found(GoalAction::INSTALL, spec, GoalJobSettings(), strict);
+        if (strict) {
+            return problem;
+        } else {
+            return GoalProblem::NO_PROBLEM;
+        }
+    }
+    auto allowed_package_types = settings.resolve_group_package_types(cfg_main);
+    auto pkg_settings = GoalJobSettings();
+    pkg_settings.with_provides = false;
```

Comment:
What about to add TODO to inherit more settings from `GoalJobSettings & settings`? Additionally it would be nice to set no globs.

Reply:
Aren't globs switched off by default? I mean the default comparator is `EQ`, which does not take globs in account.

Reply:
Correct, I am sorry I forget default value in the method. Then the implementation is completely OK. I apology for the noise.